app/src/main/java/com/arctouch/publictransportation/activity/DepartureListActivity.java

```diff
@@ -0,0 +1,53 @@
+package com.arctouch.publictransportation.activity;
+
+import android.os.Bundle;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
+
+import com.arctouch.publictransportation.R;
+import com.arctouch.publictransportation.domain.Route;
+import com.arctouch.publictransportation.fragment.DepartureListFragment;
+
+public class DepartureListActivity extends AppCompatActivity {
+
+    private Route mRoute;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_departure_list);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.detailToolbar);
+        setSupportActionBar(toolbar);
+
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        mRoute = (Route) getIntent().getSerializableExtra(DepartureListFragment.ROUTE);
+
+        if (savedInstanceState == null) {
+            Bundle arguments = new Bundle();
+            arguments.putSerializable(DepartureListFragment.ROUTE, mRoute);
+            arguments.putBoolean(DepartureListFragment.SHOW_ROUTE_DETAILS, true);
+
+            DepartureListFragment fragment = new DepartureListFragment();
+            fragment.setArguments(arguments);
```

Comment:
Setting up a Fragment's arguments on its creator is a bad idea most of the time - you leave the control of its dependencies scattered throughout the code. It's usually much better to instead create a Factory Method in the fragment class and use that instead, so these five lines of code would become simply:

```
DepartureListFragment fragment = DepartureListFragment.create(mRoute, true);